client/verta/verta/deployment/_deployedmodel.py

```diff
@@ -233,99 +207,136 @@ def predict(
+            120 seconds.
+        Returns
+        -------
+        id, prediction: Tuple(str, list)
+            The first element of the tuple is the prediction ID.
+            The second element of the tuple is the output returned by the deployed model for `x`.
+
+        Raises
+        ------
+        RuntimeError
+            If the deployed model encounters an error while running the prediction.
+        requests.HTTPError
+            If the server encounters an error while handing the HTTP request.
+
+        """
+        # Update retry config and re-mount it to the existing session if it differs from the current one.
+        self._session = http_session_utils.alter_retry_config(self._session,
+                                                              max_retries=max_retries,
+                                                              status_forcelist=retry_status,
+                                                              backoff_factor=backoff_factor)
+
+        if prediction_id:
+            http_session_utils.update_session_headers(self._session, {'verta-request-id': prediction_id})
             response = self._predict(x, compress)
+            http_session_utils.update_session_headers(self._session, {'verta-request-id': prediction_id}, remove=True)
+        else:
+            response = self._predict(x, compress)
+        id = response.headers['verta-request-id']
+        return (id, _utils.body_to_json(response))
 
-            if response.ok:
-                # This is the only part that changes
-                data = _utils.body_to_json(response)
-                headers = response.headers
-                id = headers["verta-request-id"]  # TODO: accept ID from the user
-                return id, data
-            elif response.status_code in (
-                400,
-                502,
-            ):  # possibly error from the model back end
-                try:
-                    data = _utils.body_to_json(response)
-                except ValueError:  # not JSON response; 502 not from model back end
-                    pass
-                else:  # from model back end; contains message (maybe)
-                    # try to directly print message, otherwise line breaks appear as '\n'
-                    msg = data.get("message") or json.dumps(data)
-                    raise RuntimeError(
-                        "deployed model encountered an error: {}".format(msg)
-                    )
-            elif not (
-                response.status_code >= 500 or response.status_code in (404, 429)
-            ):  # clientside error
-                break
-
-            sleep = 0.3 * (2 ** (num_retries + 1))
-            print(
-                "received status {}; retrying in {:.1f}s".format(
-                    response.status_code, sleep
-                )
+
+    # TODO: Remove this method after release of 0.22.0
```

Comment:
I personally really like well documented docstrings like these ones. Makes the API spec easier to consume and less burdensome for folks not dealing with it on the day to day. Do you think we could keep them somewhere?

Reply:
The info in this function's docsting is covered in `DeployedModel`'s class docstring!

https://github.com/VertaAI/modeldb/blob/main/client/verta/verta/deployment/_deployedmodel.py#L26-L63

Reply:
Ugh I wish our docs theme made deprecation notices more eye-catching.

> should we strip out all the info and just leave the deprecation message?

I like this idea.